server/impl/src/main/java/com/walmartlabs/concord/server/org/secret/SecretDao.java

```diff
@@ -113,7 +118,9 @@ public UUID insert(DSLContext tx, UUID orgId, UUID projectId, String name, UUID
                         SECRETS.ENCRYPTED_BY,
                         SECRETS.STORE_TYPE,
                         SECRETS.VISIBILITY)
-                .values(name, type.toString(), orgId, projectId, ownerId, encryptedBy.toString(), storeType, visibility.toString())
+                .values(name, type.toString(), orgId, projectId, ownerId, encryptedBy.toString(), storeType, visibility.toString());
+        InsertReturningStep<SecretsRecord> returningStep = (insertMode == InsertMode.UPSERT) ? builder.onDuplicateKeyIgnore() : builder;
```

Comment:
Shouldn't it be `onDuplicateKeyUpdate`?

Reply:
I tried that first, but in fact, this record is created and then the secret data is set in a separate query.  https://github.com/walmartlabs/concord/blob/master/server/impl/src/main/java/com/walmartlabs/concord/server/org/secret/SecretManager.java#L499